From afba9e1ac4c44e8c890cc2c61ae6aee9e9acdd90 Mon Sep 17 00:00:00 2001
From: He Sheng <hesheng@wxiat.com>
Date: Fri, 21 Oct 2022 15:13:08 +0800
Subject: [PATCH] sw64: make struct cpuinfo_sw64 cache line aligned
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: afba9e1ac4c44e8c890cc2c61ae6aee9e9acdd90
Modified-by-SEL: No


Sunway inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I645MA

--------------------------------

This will avoid false sharing between different CPUs.

Signed-off-by: He Sheng <hesheng@wxiat.com>
Reviewed-by: Cui Wei <cuiwei@wxiat.com>
Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/include/asm/hw_init.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/sw_64/include/asm/hw_init.h b/arch/sw_64/include/asm/hw_init.h
index 1fd7ed18c3f0..24de8977e2a7 100644
--- a/arch/sw_64/include/asm/hw_init.h
+++ b/arch/sw_64/include/asm/hw_init.h
@@ -24,7 +24,7 @@ struct cpuinfo_sw64 {
 	struct cache_desc dcache; /* Primary D or combined I/D cache */
 	struct cache_desc scache; /* Secondary cache */
 	struct cache_desc tcache; /* Tertiary/split secondary cache */
-} __attribute__((aligned(64)));
+} __aligned(SMP_CACHE_BYTES);
 
 struct cpu_desc_t {
 	__u8 model;
-- 
2.33.0

